lib/_http_common.js

```diff
@@ -212,7 +212,9 @@ function freeParser(parser, req, socket) {
     parser.outgoing = null;
     parser[kOnExecute] = null;
     if (parsers.free(parser) === false) {
-      parser.close();
+      // Make sure the parser's stack has unwound before deleting the
+      // corresponding C++ object through .close().
+      setImmediate((parser) => parser.close(), parser);
```

Comment:
It's trivial, but can you make this a top level function instead of allocating the closure?

Reply:
Does it make much of a difference if it’s not a closure? I would expect engines to optimize that away?

Reply:
Like, yes, I can do that, I’m just curious whether it makes any difference.

Reply:
Yeah, it makes a difference. In this case it would likely be trivial but it definitely has an impact. The engines do not optimize away the closures at this point